src/libImaging/Geometry.c

```diff
@@ -898,35 +783,28 @@ ImagingScaleAffine(
         xo += a[0];
     }
 
-#define AFFINE_SCALE(pixel, image)                          \
-    for (y = y0; y < y1; y++) {                             \
-        int yi = COORD(yo);                                 \
-        pixel *in, *out;                                    \
-        out = imOut->image[y];                              \
-        if (fill && x1 > x0) {                              \
-            memset(out + x0, 0, (x1 - x0) * sizeof(pixel)); \
-        }                                                   \
-        if (yi >= 0 && yi < imIn->ysize) {                  \
-            in = imIn->image[yi];                           \
-            for (x = xmin; x < xmax; x++) {                 \
-                out[x] = in[xintab[x]];                     \
-            }                                               \
-        }                                                   \
-        yo += a[4];                                         \
-    }
+    pixelsize = imOut->pixelsize;
 
     ImagingSectionEnter(&cookie);
-
-    if (imIn->image8) {
-        AFFINE_SCALE(UINT8, image8);
-    } else {
-        AFFINE_SCALE(INT32, image32);
+    for (y = y0; y < y1; y++) {
+        int yi = COORD(yo);
+        UINT8 *in, *out;
+        out = (UINT8 *)imOut->image[y];
+        if (fill && x1 > x0) {
+            memset(out + x0 * pixelsize, 0, (x1 - x0) * pixelsize);
+        }
+        if (yi >= 0 && yi < imIn->ysize) {
+            in = (UINT8 *)imIn->image[yi];
+            for (x = xmin; x < xmax; x++) {
+                for (b = 0; b < pixelsize; b++) {
+                    out[x * pixelsize + b] = in[xintab[x] * pixelsize + b];
+                }
+            }
+        }
+        yo += a[4];
     }
-
     ImagingSectionLeave(&cookie);
 
-#undef AFFINE_SCALE
-
     free(xintab);
```

Comment:
To be clear, changes like [this](https://github.com/python-pillow/Pillow/commit/5b94bf1ce863fa7d319e2fe6c8f7505f6bb5ab45) in your PR have nothing to do the fact that you've changed the pixelsize of LA/La/PA, right? You're just tidying up while you're in this general area?

In this case, your change casts the image32 pointers into UINT8 instead of INT32, resulting in this new warning when I compile it.
```
src/libImaging/Geometry.c:983:50: warning: cast to 'unsigned char *' from smaller integer type 'char' [-Wint-to-pointer-cast]
                        out[x * pixelsize + b] = (UINT8 *)imIn->image[yin][xin * pixelsize + b];
                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/libImaging/Geometry.c:983:48: warning: incompatible pointer to integer conversion assigning to 'unsigned char' from 'unsigned char *'; dereference with * [-Wint-conversion]
                        out[x * pixelsize + b] = (UINT8 *)imIn->image[yin][xin * pixelsize + b];
                                               ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
No, there is a change here. Previously it only supported `UINT8` and `INT32`, neither of which is `UINT16`. I changed it so that it always uses `UINT8` now, and uses the image pixel size to know how many bytes to copy. The diff doesn't line things up correctly, but compare for example:
`memset(out + x0, 0, (x1 - x0) * sizeof(pixel));`
`memset(out + x0 * pixelsize, 0, (x1 - x0) * pixelsize);`
Since it always uses `UINT8` now, it didn't need to be a macro. That's unfortunately what confuses the diff.

That error is valid though; I just pushed a fix.